robosuite/wrappers/domain_randomization_wrapper.py

```diff
@@ -0,0 +1,218 @@
+"""
+This file implements a wrapper for facilitating domain randomization over
+robosuite environments.
+
+Categories of DR:
+
+    Observation Noise
+
+        All
+
+            Sensor Dropout
+
+                TODO
+
+        RGB Image
+
+            Texture Randomization
+                Color (Relative / Absolute)
+                Pattern
+
+            Camera Randomization
+                Position (Relative)
+                Rotation (Relative)
+
+            Lighting Randomization
+                Position (Relative)
+                Direction (Relative)
+                Color (Relative)
+                    Specular
+                    Ambient
+                    Diffuse
+                Active
+                CastShadow
+                    TODO: do we really want this? excluding for now...
+
+        Low-Dimension
+
+            Pose Noise
+            Joint Noise
+            Velocity Noise
+
+            TODO (requires observation categorization, too much of a pain for now...)
+                -Perhaps only noise up the proprioception??? (and assume fixed set of keys)
+        
+    Action Noise
+
+        TODO
+
+    Dynamics Parameter Noise
+
+        TODO
+
+TODO: Implement Observation Dropout, Action Noise, Dynamics Randomization
+TODO: why can we still see things with all the lights off?
+TODO: each composite object should specify texture groups that get randomized together
+    (geom groups per texture)
+TODO: color perturbations could probably be improved by using a space other than RGB
```

Comment:
maybe provide an argument option to specify between HSV or RGB? should only take a couple of transform functions to be added (though propagated through all the classes)

Reply:
I wanted to make sure we have minimal functionality before adding another color space - I think RGB appeared to work fine from my tests. I'd prefer we see a need for other color spaces before we put the effort to include them.